crates/steel-core/src/primitives/ports.rs

```diff
@@ -1,13 +1,17 @@
-use crate::rvals::{Result, SteelString, SteelVal};
+use crate::rvals::{RestArgsIter, Result, SteelByteVector, SteelString, SteelVal};
 use crate::steel_vm::builtin::BuiltInModule;
 use crate::stop;
 use crate::values::port::new_rc_ref_cell;
 use crate::values::port::{SteelPort, SteelPortRepr};
+use crate::values::structs::{make_struct_singleton, StructTypeDescriptor};
 
+use once_cell::unsync::Lazy;
 use steel_derive::function;
 
 thread_local! {
-    pub static EOF_OBJECT: SteelString = "eof".into();
+    static EOF_OBJECT: Lazy<(SteelVal, StructTypeDescriptor)>= Lazy::new(|| {
```

Comment:
:information_source: Added a proper EOF object, as a singleton of a unique struct.

Reply:
Thanks! I've been meaning to do this and just kept forgetting.

This works fine and I don't have anything against it. Did you consider something like this?:

```rust
struct EofObject;

impl Custom for EofObject {
    // probably implement the equality method?
}
```

Actually having typed that out, I think I like your solution better since we can just inherit struct equality and everything like that